vyper/parser/arg_clamps.py

```diff
@@ -0,0 +1,74 @@
+import functools
+
+from vyper.parser.lll_node import (
+    LLLnode,
+)
+from vyper.types.types import (
+    ByteArrayLike,
+    ListType,
+    get_size_of_type,
+    is_base_type,
+)
+from vyper.utils import (
+    MemoryPositions,
+)
+
+
+def _mk_calldatacopy_copier(pos, sz, mempos):
+    return ['calldatacopy', mempos, ['add', 4, pos], sz]
+
+
+def _mk_codecopy_copier(pos, sz, mempos):
+    return ['codecopy', mempos, ['add', '~codelen', pos], sz]
+
+
+def make_arg_clamper(datapos, mempos, typ, is_init=False):
+    """
+    Clamps argument to type limits.
+    """
+
+    if not is_init:
+        data_decl = ['calldataload', ['add', 4, datapos]]
+        copier = functools.partial(_mk_calldatacopy_copier, mempos=mempos)
+    else:
+        data_decl = ['codeload', ['add', '~codelen', datapos]]
+        copier = functools.partial(_mk_codecopy_copier, mempos=mempos)
+    # Numbers: make sure they're in range
+    if is_base_type(typ, 'int128'):
+        return LLLnode.from_list([
+            'clamp',
+            ['mload', MemoryPositions.MINNUM],
```

Comment:
I've never really understood why we store these limits in memory instead of simply supplying the value at compile time. (i.e., `['clamp', SizeLimits.MINNUM, data_decl, SizeLimits.MAXNUM]`). Is there some reason we can't get rid of references to MemoryPositions.MINNUM and friends?

Reply:
I tried it at some point bit the bytecode size ran away too fast.

Reply:
Especially on the upper limits.

Reply:
Oh because it requires like push32, gotcha

Reply:
Seems like potential for future optimization

Reply:
One potential optimization is to store the constants in a datasection of the code. I think we may eventually want to do this for user-defined literals/constants as well. This preserves the bytecode and amortized gas costs but could get the memory size down 5 words :).

Another possible optimization is to make them LLL constants but that takes up some stack slots which is even more precious than memory.